File: boj_10799.py
data = list(input())
ans = 0
stk = []

for i in range(len(data)):
    if data[i] == "(":
        stk.append("(")
    else:
        if data[i - 1] == "(":
            stk.pop()
            ans += len(stk)
        else:
            stk.pop()
            ans += 1

print(ans)

File: boj_10799.cpp
#include <iostream>
#include <stack>
#include <string>
using namespace std;

int main() {
  string s;
  stack<char> stk;
  int cnt = 0;
  cin >> s;
  for (int i = 0; i < s.length(); i++) {
    if (s[i] == '(') {
      stk.push(s[i]);
    } else {
      stk.pop();
      if (s[i - 1] == '(') {
        cnt += stk.size();
      } else {
        cnt++;
      }
    }
  }
  cout << cnt << endl;
}